metrics/src/metrics.rs

```diff
@@ -280,7 +280,10 @@ impl MetricsAgent {
             counters,
         );
 
-        if !combined_points.is_empty() {
+        // `combined_points`` always contain a `metric` point for how many real
+        // datapoints are submitted at the end. Therefore, to skip empty point
+        // submission, `combined_points.len()` must be larger than 1.
+        if combined_points.len() > 1 {
             writer.write(combined_points);
         }
```

Comment:
This is actually something I was not sure about, when I was changing this code.
Is there any value is submitting metrics stats that contain zeroes?

I thought that there is, as it serves as a kind of ping, indicating that metrics are properly uploaded.  There is just nothing to upload.
While if we do not submit anything, it is impossible to distinguish a case when there are no metrics to submit, from a case when something is broken in the metrics upload process.

I should have removed the `if` statement completely, I guess.

But as you are proposing this change, there is probably a flaw or some practical issue with my approach, right?
Could you explain why are you suggesting this change?

Reply:
This change makes `test_flush_before_drop` fails intermittently on CI. 
More details can be found on discord.
 https://discord.com/channels/428295358100013066/560503042458517505/1232341710885879870

Reply:
Sorry for a slow response.
What do you think about this https://github.com/anza-xyz/agave/pull/1039 instead?